### PR TITLE
[frontend/backend] Ability to filter on representatives in streams/playbooks/triggers (#6799)

### PR DESCRIPTION
**Pull request by Archidoit**
Chunk 1 of issue: https://github.com/OpenCTI-Platform/opencti/issues/6799

Chunk 1: ability to filter on representatives in streams/playbooks/triggers

(Chunk 2 later: ability to filter on representatives in feeds, taxii and entity lists)

doc PR : https://github.com/OpenCTI-Platform/docs/pull/165

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/profile/triggers/TriggerEditionOverview.tsx [hunk 1]
```diff
@@ -469,9 +447,9 @@ const TriggerEditionOverview: FunctionComponent<TriggerEditionOverviewProps> = (
                     filters={filters}
                     helpers={helpers}
                     redirection
-                    searchContext={{ entityTypes: ['Stix-Core-Object', 'stix-core-relationship'] }}
-                  />
-                )}
+                    searchContext={{ entityTypes: ['Stix-Core-Object', 'stix-core-relationship'] }}entityTypes={['Stix-Core-Object', 'stix-core-relationship', 'Stix-Filtering']}
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/profile/triggers/TriggerEditionOverview.tsx]: looks like an issue here

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx [hunk 2]
```diff
@@ -79,6 +79,32 @@ export const contextFilters = [
   'contextObjectMarking',
 ];
 
+// filters available in stix filtering (streams, playbooks, triggers)
+export const stixFilters = [
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx:83]: nice refactoring 😌

File: opencti-platform/opencti-front/src/private/components/data/stream/StreamCollectionCreation.tsx [hunk 3]
```diff
@@ -225,6 +203,7 @@ const StreamCollectionCreation: FunctionComponent<StreamCollectionCreationProps>
                 styleNumber={2}
                 redirection
                 searchContext={{ entityTypes: ['Stix-Core-Object', 'stix-core-relationship'] }}
+                entityTypes={['Stix-Core-Object', 'stix-core-relationship', 'Stix-Filtering']}
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/data/stream/StreamCollectionCreation.tsx:206]: What is the difference between `searchContext `and `entityTypes `for `FilterIconButton` ? Sometimes we have one of them only, sometimes they are the same, sometimes not.

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx (hunk 2 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx:83]: thanks :)

File: opencti-platform/opencti-front/src/private/components/profile/triggers/TriggerEditionOverview.tsx (hunk 1 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/private/components/profile/triggers/TriggerEditionOverview.tsx]: fixed

File: opencti-platform/opencti-front/src/private/components/data/stream/StreamCollectionCreation.tsx (hunk 3 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/private/components/data/stream/StreamCollectionCreation.tsx:206]: searchContext is used to look for the available entities to display in some filter lists that display entities (exemple: creator filter) and it is used in useSearchEntities // entityTypes is the entity types whose filter keys we want to fetch from the schema

**Comment by SouadHadjiat** [comment]
Not really related to this PR, but do you know why we have two methods for representative extraction :  extractRepresentative (in entity-representative) and extractStixRepresentative (in stix-representative) ? what is the difference between the two ?

**Comment by Archidoit** [comment]
@SouadHadjiat : extractStixRepresentative extracts the representative of an entity that is in the stix format, extractRepresentative extracts the representative of an entity in the BasicStoreCommon format

**Comment by SouadHadjiat** [comment]
Can they return different values ? about the main representative.

**Comment by Archidoit** [comment]
@SouadHadjiat there are some difference. For instance, a status is not defined in stix-representative. But if things have been correctly done, they should be no difference for a given entity type.

**Comment by Archidoit** [comment]
@SouadHadjiat  there are some differences (exemple for Attack Patterns : name in stix representative // composition of x_mitre_id and name in entity representative)

**Review by SouadHadjiat — APPROVED** [review]
tested locally ✅

**Pull request merged**